realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -62,7 +62,7 @@ public DynamicRealmObject(RealmObject obj) {
 
     DynamicRealmObject(BaseRealm realm, Row row) {
         this.realm = realm;
-        this.row = (row instanceof CheckedRow) ? (CheckedRow) row : ((UncheckedRow) row).convertToChecked();
+        this.row = row instanceof CheckedRow ? (CheckedRow) row : ((UncheckedRow) row).convertToChecked();
```

Comment:
I'm not really a fan of enforcing this rule. In a lot of cases I find it increases readability IMO.
E.g here. I'll highlight a few more places where below